### PR TITLE
Add "Pack Project As..." button to menu

### PR DESCRIPTION
**Pull request by glebmish**
Also refactor setting filter for file dialog

### CODE REVIEW

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 1]
```diff
@@ -1261,6 +1274,37 @@ void MainWindow::slot_save_project_as()
...
+QString MainWindow::get_filter_string(const int filter)
+{
+    QString filter_string = "";
+
+    if (filter & MainWindow::PROJECT)
+        filter_string += "Project Files (*.appleseed ; *.appleseedz)";
+
+    if (filter & MainWindow::APPLESEED)
+    {
+        if (filter_string.size() != 0)
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: You can use `!filter_string.isEmpty()` which is more idiomatic.

File: src/appleseed.studio/mainwindow/mainwindow.ui [hunk 2]
```diff
@@ -48,7 +48,7 @@
      <x>0</x>
      <y>0</y>
      <width>1280</width>
-     <height>21</height>
+     <height>38</height>
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: Not sure why this changes?

File: src/appleseed.studio/mainwindow/mainwindow.ui [hunk 3]
```diff
@@ -162,7 +163,7 @@
   <widget class="QDockWidget" name="project_explorer">
    <property name="minimumSize">
     <size>
-     <width>200</width>
+     <width>500</width>
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: I think you'll need to revert these dimension changes.

File: src/appleseed.studio/mainwindow/mainwindow.ui [hunk 4]
```diff
@@ -306,6 +307,9 @@
    <property name="text">
     <string>Save Project &amp;As...</string>
    </property>
+   <property name="softKeyRole">
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: No idea what this property is about, I think you can safely delete it.

File: src/appleseed.studio/mainwindow/mainwindow.ui (hunk 2 above)
Comment (glebmish) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: it's automatically changed in QtDesigner when I added new button

File: src/appleseed.studio/mainwindow/mainwindow.h [hunk 5]
```diff
@@ -201,6 +201,17 @@ class MainWindow
     // Miscellaneous.
     void print_startup_information();
     virtual void closeEvent(QCloseEvent* event);
+    void do_save_project(const int filter);
+    QString get_filter_string(const int filter);
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.h]: Make this method static.

File: src/appleseed.studio/mainwindow/mainwindow.ui (hunk 3 above)
Comment (glebmish) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: QtDesigner change it as well

File: src/appleseed.studio/mainwindow/mainwindow.ui (hunk 4 above)
Comment (glebmish) [inline src/appleseed.studio/mainwindow/mainwindow.ui]: QtDesigner change it as well

File: src/appleseed.studio/mainwindow/mainwindow.cpp [hunk 6]
```diff
@@ -1261,6 +1274,25 @@ void MainWindow::slot_save_project_as()
     }
 }
 
+QString MainWindow::get_filter_string(const int filter)
+{
+    QStringList filters;
+
+    if (filter & MainWindow::PROJECT)
+        filters << "Project Files (*.appleseed ; *.appleseedz)";
```
Comment (dictoon) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: Should be

    "Project Files (*.appleseed;*.appleseedz)"

Comment (glebmish) [inline src/appleseed.studio/mainwindow/mainwindow.cpp]: It doesn't work this way, I checked 
Only works with spaces
